CMS-Enterprise/sbom-harbor
# feat(gh-prov-refactor-cdx-commit): removed external mongo service in Github Service and refactored GitHub Provider to use commit hashes as database key for Commits instead of the URL of the repository

## Summary

Refactored GitHub Provider and service to remove the external Mongo Service and made the Commit's DB key the last hash rather than the URL.  This caused a fairly significant refactor.

### Changed

- The Commit DB key to the last hash in the repo

### Removed

- The external Mongo Service in the GitHub Provider. Moved functionality into the GitHub service proper.

## How to test

1. Set up test environment
2. Run `cargo test`
3. Run `./target/debug/harbor ingest -p github harbor-test-org --debug`
  - Verify that all of the repos are processed as usual
  - Run the command a second time and verify that all of the commit hashes extracted from the previous run are 
    found and no repos are processed redundantly.
  - Extras credit: Change a file in one of the repos under the `harbor-test-org`, commit and push the change. 
    Run the harbor command again and verify that _only_ that repo is processed. 
4. Open Compass and look at the data:  
    - In the `Commit` collection, the `id` will now be a string identifiable as a Git Commit hash rather than the URL.
    - In the `Package` collection, the results will be the same as past runs
